include/RED4ext/NativeTypes-inl.hpp

```diff
@@ -100,3 +101,145 @@ RED4EXT_INLINE bool RED4ext::TweakDBID::operator!=(const TweakDBID& aDBID) const
 {
     return !(aDBID == *this);
 }
+
+RED4EXT_INLINE RED4ext::Variant::Variant(const RED4ext::CBaseRTTIType* aType)
+    : Variant()
+{
+    if (aType)
+    {
+        Init(aType);
+    }
+}
+
+RED4EXT_INLINE RED4ext::Variant::Variant(const RED4ext::CBaseRTTIType* aType, const RED4ext::ScriptInstance aData)
+    : Variant()
+{
+    if (aType)
+    {
+        Fill(aType, aData);
+    }
+}
+
+RED4EXT_INLINE RED4ext::Variant::Variant(const RED4ext::CName& aTypeName, const RED4ext::ScriptInstance aData)
+    : Variant(RED4ext::CRTTISystem::Get()->GetType(aTypeName), aData)
+{
+}
+
+RED4EXT_INLINE RED4ext::Variant::Variant(const Variant& aOther)
+    : Variant(aOther.GetType(), aOther.GetDataPtr())
+{
+}
+
+RED4EXT_INLINE RED4ext::Variant::~Variant()
+{
+    Free();
+}
+
+RED4EXT_INLINE bool RED4ext::Variant::IsEmpty() const noexcept
```

Comment:
Maybe we should have an operator or both an operator and this? What do you think?

Reply:
I'd keep named function for internal use, but perhaps rename it to `IsValid` or `IsDefined`.
I only skipped the operators for now because I wanted to think more about use cases when working on examples.
We can also add some helpers and operators for `IScriptable` types as they come with type info.

Reply:
👍 for now let's keep it like this.